robottelo/ui/computeresource.py

```diff
@@ -155,3 +153,87 @@ def search_container(self, cr_name, container_name):
             locators['resource.filter_containers'], container_name)
         strategy, value = locators['resource.select_container']
         return self.wait_until_element((strategy, value % container_name))
+
+    def list_vms(self, res_name):
+        """Lists vms of a particular compute resource.
+
+        Note: Currently lists only vms that show up on the first page.
+        """
+        self.click(self.search(res_name))
+        self.click(tab_locators['resource.tab_virtual_machines'])
+        vm_elements = self.find_elements(locators['resource.vm_list'])
+        return [vm.text for vm in vm_elements]
+
+    def add_image(self, res_name, parameter_list):
+        """Adds an image to a compute resource."""
+        if parameter_list is None:
+            return
+        self.click(self.search(res_name))
+        self.click(locators['resource.image_add'])
+        self.wait_until_element(locators['resource.image_name'])
+        for parameter_name, parameter_value, parameter_type in parameter_list:
+            param_locator = '.'.join((
+                'resource.image',
+                (parameter_name.lower()).replace(' ', '_')
+            ))
+            if parameter_type == 'field':
```

Comment:
It doesn't seem that we need `parameter_type` logic anymore as we have `assign_value` method, so next statement should work for us:
`self.assign_value(locators[param_locator], parameter_value)`